app/models/project/phase_definition.rb

```diff
@@ -26,42 +28,28 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
-class Project::LifeCycleStepDefinition < ApplicationRecord
+class Project::PhaseDefinition < ApplicationRecord
   include ::Scopes::Scoped
 
-  has_many :life_cycle_steps,
-           class_name: "Project::LifeCycleStep",
+  has_many :phases,
+           class_name: "Project::Phase",
            foreign_key: :definition_id,
            inverse_of: :definition,
            dependent: :destroy
-  has_many :projects, through: :life_cycle_steps
+  has_many :projects, through: :phases
   belongs_to :color, optional: false
 
   validates :name, presence: true, uniqueness: true
-  validates :type, inclusion: { in: %w[Project::StageDefinition Project::GateDefinition], message: :must_be_a_stage_or_gate }
-  validate :validate_type_and_class_name_are_identical
-
-  attr_readonly :type
-
+  validates :start_gate_name, presence: true, if: Proc.new { |d| d.start_gate.present? }
+  validates :end_gate_name, presence: true, if: Proc.new { |d| d.end_gate.present? }
```

Comment:
@dombesz 
`d.start_gate.present?` is same as `d.start_gate?` and validates allows to use a Symbol, so:
```suggestion
  validates :start_gate_name, presence: true, if: :start_gate?
  validates :end_gate_name, presence: true, if: :end_gate?
```